### ISSUE TITLE
Add a `BindLeft` type class

### ISSUE THREAD

**Issue by natefaubion** [purescript/purescript-bifunctors#12]
```purescript
class BindLeft m where
  lbind :: forall a b r. m a r -> (a -> m b r) -> m b r

class (Biapplicative m, BindLeft m) <= MonadLeft m
```

This is a generalized, type-changing `catch` operation, which is useful for things like `Either`, and also lets you implement a `Monad` instance for `Flip`.

**Comment by JordanMartinez** [comment]
Wouldn't you also need to define an `ApplyLeft` and `ApplicativeLeft` to enable `Flip` to have a `MonadLeft` instance?

**Comment by natefaubion** [comment]
@JordanMartinez Yes, my formulation above is incorrect. Biapplicative isn't appropriate for this.

**Comment by mikesol** [comment]
Ran across this today as I was looking for a `bindLeft` in a project. Would be useful!

**Comment by JordanMartinez** [comment]
Maybe this would be better moved to the 'qualified do' package?
